### ISSUE TITLE
If Track Active Item in Solution Explorer is ON, GitBranchDiffer opens diff window even if user changes document

### ISSUE THREAD

**Issue by sajalverma17** [sajalverma17/GitBranchDiffer#4]
**Version Information :**
 - Visual Studio 2019 Pro
 - Git Branch Differ Version - Pre-release

**Describe the bug**
Bug apprears when opening a file from Solution Explorer after the "Show diff in solution" filter was successfully applied on in.
If user changes the document window in documents pane, and this triggers a change in solution explorer, branch diff window is opened.

**To Reproduce**
1. Options -> General -> Check "Track Active Item in Solution Explorer"
2. Apply GitBranchDiff filter
3. Change documents to another document which exists in filtered Solution Explorer  
4.  This opens a the document, but also opens the comparison of the file with base branch, even if user did not click the item in Solution Explorer

**Expected behavior**
GitBranchDiffer should only open a branch diff window if they are selecting a node in solution explorer. Any other changes should not result in this

**Comment by sajalverma17** [comment]
This bug exists because we use the Selection Changed DTE events, and use them to see if anything selection was changed in solution explorer, and then open a comparison window.

Ideally, we should have used Document DTE events to open a comparison window. That is leads to another bug and I switched to selection changed events. We can not suppress VS from opening the document in regular view and force it to open comparison view when using Document Events. So I ended up with Selection Changed DTE event so a comparison window opens if user click an item in Solution Explorer. 

This bug is a drawback of that approach and I may have to find a third approach that fixes both the bugs.
